pages/_app.js

```diff
@@ -1,10 +1,36 @@
+import { useState } from "react";
 import GlobalStyle from "../styles";
+import { initialPlants } from "@/assets/plants";
 
 export default function App({ Component, pageProps }) {
+  const [plants, setPlants] = useState(initialPlants);
+  const [plantsInfo, setPlantsInfo] = useState(
+    plants.map((plant) => ({ ...plant, isMyPlant: false }))
+  );
+
+  const handleToggleMyPlants = (plant) => {
+    const updatedPlantsInfo = plantsInfo.map((info) =>
+      info.id === plant.id ? { ...info, isMyPlant: !info.isMyPlant } : info
+    );
+
+    setPlantsInfo(updatedPlantsInfo);
+  };
+
+  const isMyPlantfunction = (plant) => {
+    const plantInfo = plantsInfo.find((info) => info.id === plant.id);
+    return plantInfo ? plantInfo.isMyPlant : false;
+  };
   return (
     <>
       <GlobalStyle />
-      <Component {...pageProps} />
+      <Component
+        {...pageProps}
+        plants={plants}
+        plantsInfo={plantsInfo}
+        setPlantsInfo={setPlantsInfo}
```

Comment:
passing a setter function as a prop is not a good practice (you give away the complete control over the state). Better: Define a detailed handler function (which sets the state) and pass the handler function. (Like you do with handleToggleMyPlants)